what4/src/What4/Protocol/Online.hs

```diff
@@ -327,6 +363,17 @@ tryPop p =
 inNewFrame :: (MonadIO m, MonadMask m, SMTReadWriter solver) => SolverProcess scope solver -> m a -> m a
 inNewFrame p action = inNewFrameWithVars p [] action
 
+-- For abduction, we want the final assertion to be a in a new frame, so that it 
+-- can be closed before asking for abducts. The following two commands allow frame 2 
+-- to be pushed and popped independently of other commands
+-- | Open a second solver assumption frame.
```

Comment:
It would be worth moving the `-- | Open a second solver assumption frame.` part to the top of the comments so that the commentary about abduction is included in the Haddocks.